include/parallel/parallel_sync.h

```diff
@@ -100,6 +100,34 @@ void pull_parallel_vector_data(const Communicator & comm,
                                ActionFunctor & act_on_data,
                                const datum * example);
 
+/**
+* Send and receive and act on vectors of data. Similar to
+* push_parallel_vector_data, except the vectors are packed and unpacked
+* using the Parallel::Packing routines.
```

Comment:
We don't actually want to use vector inputs in this case, do we?  There's no performance benefit since we're serializing into buffers anyway, and working with iterator ranges is a lot more flexible.

Reply:
I think this is reasonable, but might be a little more than I can wrap my head around. I believe @friedmud wanted to add some input to this